src/core/ext/client_channel/client_channel.c

```diff
@@ -543,7 +543,10 @@ static grpc_error *cc_init_channel_elem(grpc_exec_ctx *exec_ctx,
                                         grpc_channel_element *elem,
                                         grpc_channel_element_args *args) {
   channel_data *chand = elem->channel_data;
-  GPR_ASSERT(args->is_last);
+  memset(chand, 0, sizeof(*chand));
+  if (!args->is_last) {
```

Comment:
This should stay an assert: there should be no way from API to get to the point that client_channel is not the last thing in the stack, and having it be elsewhere is going to cause bugs.